pkg/interactor/data.go

```diff
@@ -56,6 +56,22 @@ func (d *DataInteractor) Save(data Data) error {
 	return err
 }
 
+func (d *DataInteractor) DeleteByDeviceID(deviceId string) error {
+	err := d.DataStore.Delete(deviceId)
+	if err != nil {
+		d.logger.Error(err)
+	}
+	return err
+}
+
+func (d *DataInteractor) DeleteAll() error {
+	err := d.DataStore.Delete("")
+	if err != nil {
+		d.logger.Error(err)
+	}
+	return err
+}
+
```

Comment:
Error handling should be done right after checking for errors (inside the `if` statement). If no errors occur, we return `nil` instead. 

In this case, the functions are too small and it might not be trivial to understand the reason for the suggestion below. But imagine this:

1. `ControllerFoo` calls `InteractorMethod()`
2. `InteractorMethod()` then calls `functionFoo()` and `functionBar()`
3. `InteractorMethod()` handles returned errors for `functionFoo()` and `functionBar()`, as below:
```Go
func (i *interactor) InteractorMethod() {
  err := functionFoo()
  if err != nil {
    d.logger.Error(err)
  }
  err = functionBar()
  if err != nil {
    d.logger.Error(err)
  }
  return err
}
```
4. `ControllerFoo` (which called `InteractorMethod()`) would then receive the error from `functionBar()` instead of the primal error, which happened in `functionFoo()`.
5. `ControllerFoo` will then wrongfully handle the error, as well as we have added overhead (additional processing/complexity) unnecessarily, as we continued to run the subsequent instructions inside `InteractorMethod()` when we shouldn't have.

Right now, the code is quite simple and we wouldn't see many problems. However, as the codebase scales and we add functionalities further in the project, this could become a problem. To avoid this, I suggest we standardize error handling to be done within it's error-checking scope and return `nil` instead of `err` :point_down: 
```suggestion
func (d *DataInteractor) DeleteByDeviceID(deviceId string) error {
	err := d.DataStore.Delete(deviceId)
	if err != nil {
		d.logger.Error(err)
		return err
	}
	return nil
}

func (d *DataInteractor) DeleteAll() error {
	err := d.DataStore.Delete("")
	if err != nil {
		d.logger.Error(err)
		return err
	}
	return nil
}
```

Reply:
Done!